### PR TITLE
Add e2e job in openshift/origin for CSI

### PR DESCRIPTION
**Pull request by bertinatto**
CC @openshift/storage

Please see commit messages for additional details.

### CODE REVIEW

**Comment by bertinatto** [comment]
/hold
I need to check if everything goes well before merging.

**Comment by bertinatto** [comment]
/hold cancel

The jobs I'm adding here should be optional. i.e., they shouldn't block PRs from merging if they fail.

CC @openshift/storage 

@bbguimaraes I'd appreciate if you could take a look as well. Thanks!

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 1]
```diff
@@ -239,6 +239,12 @@ tests:
   commands: TEST_SUITE=openshift/conformance/serial run-tests
   openshift_installer:
     cluster_profile: aws
+- as: e2e-aws-csi
+  commands: TEST_INSTALL_CSI_DRIVERS=aws-ebs TEST_SUITE=openshift/csi run-tests
+  optional: true
+  skip_report: true
```
Comment (bbguimaraes) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: Sorry, I should have been clearer: these should be in the jobs (under `ci-operator/jobs`), not here. `prowgen` will not overwrite them when you regenerate.

Comment (bertinatto) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml]: Will fix, thanks!

**Comment by bertinatto** [comment]
/hold
to fix the `optional` option

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml (hunk 1 above)
Comment (bertinatto) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml]: Fixed, PTAL.

I left the `optional` and `skip_report` options in a separate commit (https://github.com/openshift/release/pull/7131/commits/24afd2914adedb30f4fae12264091494b18a3bc9) to make the review easier.

**Comment by bertinatto** [comment]
/hold cancel

1. Added the jobs we needed in `ci-operator/config/openshift/origin/openshift-origin-*`
2. Ran `make jobs`
3. Added the `optional` and `skip_report` options to files in `ci-operator/config/openshift/origin/openshift-origin-*`

**Comment by bertinatto** [comment]
`could not wait for build: the build machine-os-content failed after 7m31s with reason DockerBuildFailed: Docker build strategy has failed.`

**Comment by bertinatto** [comment]
`could not wait for build: the build machine-os-content failed after 10m18s with reason DockerBuildFailed: Docker build strategy has failed.`

**Comment by bertinatto** [comment]
`could not wait for build: the build machine-os-content failed after 10m18s with reason DockerBuildFailed: Docker build strategy has failed.`

/test pj-rehearse

**Comment by bertinatto** [comment]
/test pj-rehearse

```
2020/02/19 12:25:21 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/02/19 12:25:22 failed to release leaked leases ([53162864-5217-47c3-b07a-f56cf02c18e1]): no resource name 53162864-5217-47c3-b07a-f56cf02c18e1
2020/02/19 12:25:22 Ran for 1h26m9s
2020/02/19 12:25:23 Submitted failure event to sentry (id=437c11328f5d41d797eea3fe86defce7)
error: could not run steps: step e2e-aws-csi failed: status 503 Service Unavailable, statusCode 503 releasing 53162864-5217-47c3-b07a-f56cf02c18e1
```

**Comment by bertinatto** [comment]
/test pj-rehearse

```
failed to acquire lease: resources not found
```

**Comment by bertinatto** [comment]
/test pj-rehearse

```
level=info msg="Cluster operator openshift-apiserver Available is False with APIServerDaemonSet_NoAPIServerPod::APIServices_PreconditionNotReady: APIServicesAvailable: PreconditionNotReady\nAPIServerDaemonSetAvailable: no openshift-apiserver daemon pods available on any node."
level=error msg="Cluster operator support Degraded is True with UploadFailed: Unable to report: Post https://cloud.redhat.com/api/ingress/v1/upload: dial tcp: i/o timeout"
level=info msg="Cluster operator support Disabled is False with : "
level=info msg="Pulling debug logs from the bootstrap machine"
level=info msg="Bootstrap gather logs captured here \"/tmp/artifacts/installer/log-bundle-20200219194141.tar.gz\""
level=fatal msg="Bootstrap failed to complete: failed to wait for bootstrapping to complete: timed out waiting for the condition"
```

**Pull request merged**